cuda/base/executor.cpp

```diff
@@ -70,6 +69,53 @@ class device_guard {
     int original_device_id{};
 };
 
+
+// The function is copied from _ConvertSMVer2Cores of
+// cuda-9.2/samples/common/inc/helper_cuda.h
```

Comment:
Good thinking, that is the best place to get that sort of function from. Sadly they don't provide anything directly in their API...